src/pages/thank-you.js

```diff
@@ -4,27 +4,68 @@ import Signup from '../components/signup.js';
 import Social from '../components/social.js';
 import ThankYouHeader from '../components/thank-you-header.js';
 import analytics from '../lib/analytics.js';
+import locationSearchParser from '../lib/locationSearchParser.js';
+import amountModifier from '../lib/amount-modifier.js';
+import suggestMonthly from '../lib/suggest-monthly.js';
+
+/*
+
+  Stripe calls this route as:
+
+  /thank-you/? payment = Stripe
+              &str_amount = number in "cents"
+              &str_currency = currency id string
+              &str_id = transaction id string
+              &str_frequency = "one-time" or "monthly"
+              &email = email string
+              &country = full country name
+
+  Paypal calls this route as:
+
+  ...
+
+*/
 
 var ThankYou = React.createClass({
   contextTypes: {
     intl: React.PropTypes.object
   },
+  componentWillMount: function() {
+    this.monthlyInvitation = null;
+
+    let query = locationSearchParser(window.location);
```

Comment:
Doesn't this break on the server side render?

Reply:
it does not - I expect because of the polyfills.